query/graphql/src/main/java/org/codice/ddf/admin/graphql/transform/GraphQLTransformOutput.java

```diff
@@ -185,6 +186,12 @@ public Object functionDataFetcher(DataFetchingEnvironment env, FunctionField<Fie
 
         if (!result.getErrorMessages()
                 .isEmpty()) {
+
+            //Hide the Hidden Fields
+            funcField.getArguments().stream()
```

Comment:
This is the part that is sort of confusing me. We already coerce `Field`s on their way out if they have the `HIDDEN_FIELD` type. Do we need hide things here as well? Are we double hiding fields?

Reply:
From my understanding, we are hiding it here for the logs because right after this line we are throwing a `FunctionDataFetcherException` which will do a `getValue()` on all the fields and display them.

Reply:
Not in love with this. It's not scalable to other types of fields that need to hide their data in some way or another (we may never have one, but I wouldn't state that with any certainty) and it's very leaky. 

I think you can do this:
1. Add this method to `Field`: 
```java
    default T getSanitizedValue() {
        return getValue();
    }
```
2. Get rid of the `hide()` method and all its uses.
3. Add this to `BaseHiddenField` in its place:
```java
    @Override
    public String getSanitizedValue() {
        return HIDDEN_FLAG;
    }
```
4. In places where you need to hide _anything_ (today, a hidden field; tomorrow something else), you call get `getSanitizedValue()` method instead of `getValue()`. In places where you still need the real value, you call that.

I _think_ that will work for all cases.